modules/javafx.web/src/main/native/Source/WebCore/platform/java/FileSystemJava.cpp

```diff
@@ -232,21 +232,76 @@ String openTemporaryFile(const String&, PlatformFileHandle& handle)
     return String();
 }
 
-PlatformFileHandle openFile(const String&, FileOpenMode)
+PlatformFileHandle openFile(const String& path, FileOpenMode mode)
 {
-    notImplemented();
-    return invalidPlatformFileHandle;
+    JNIEnv* env = WebCore_GetJavaEnv();
+    static jmethodID mid = env->GetStaticMethodID(
+            GetFileSystemClass(env),
+            "fwkOpenFile",
+            "(Ljava/lang/String;Ljava/lang/String;)Ljava/io/RandomAccessFile;");
+    ASSERT(mid);
+
+    // fileMode initialized to NULL to avoid compilation warning
+    jstring fileMode = NULL;
+    if (mode == FileOpenMode::Read) {
+        fileMode = env->NewStringUTF("r");
+    } else if (mode == FileOpenMode::Write) {
+        fileMode = env->NewStringUTF("rw");
+    }
+    ASSERT(fileMode);
+
+    PlatformFileHandle result = env->CallStaticObjectMethod(
+            GetFileSystemClass(env),
+            mid,
+            (jstring)path.toJavaString(env), fileMode);
+
+    result = env->NewGlobalRef(result);
```

Comment:
Use JGObject instead of explicit ref counting APIs.

Also it is better to define PlatformFileHandle as JGObject. It will automatically call env->DeleteGlobalRef when WebCore releases PlatformFileHandle

`typedef JGObject PlatformFileHandle;`